modules/event-emitter/main/index.coffee

```diff
@@ -74,19 +74,23 @@ class EventEmitter
     ee.emitters.add(be)
     be
 
-  removeEmitter = (ee, namespace) ->
-    if ee.emittersMap.has(namespace)
+  removeEmitter = (ee, be, namespace) ->
+    if namespace and ee.emittersMap.has(namespace)
       ee.emittersMap.delete(namespace)
-      ee.emitters.remove(ee)
-    ee
+    else
+      lookedUpNamespace = ee.emittersMap.entries().filter(([_, e]) -> e is be)[0]?[0]
+      if lookedUpNamespace
+        ee.emittersMap.delete(lookedUpNamespace)
+    ee.emitters.remove(be)
 
   # emit an object to all callbacks registered with the name given
   emit: (name, data) ->
     if not @suppressedMap.get(name)
-      if @deprecatedEvents?
-        for e of @deprecatedEvents
-          if @deprecatedEvents[e].event is name
-            @emit e, data
+      # XXX: Deprecated event check - This is useful to have if we need to deprecated events in the future
+      # if @deprecatedEvents?
+      #   for e of @deprecatedEvents
+      #     if @deprecatedEvents[e].event is name
+      #       @emit e, data
```

Comment:
The other part of the deprecated event check is commented out so I commented this out too